From: Pawel Zalewski <pawel.zalewski@thecapablehub.orgg>
Date: Mon, 23 Feb 2026 13:56:22 +0000
Subject: [PATCH] Makefile: use MUSL_LDSO_ARCH variable across the stack

This variable is used in Yocto to derive the path for the ld, lets
just use a single way to define all variables, including the linker
itself.

Upstream-Status: Inappropriate [Yocto specific]
Signed-off-by: Pawel Zalewski <pawel.zalewski@thecapablehub.org>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index abae4f73..13d7f76f 100644
--- a/Makefile
+++ b/Makefile
@@ -75,7 +75,7 @@ ALL_TOOLS = obj/musl-gcc
 WRAPCC_GCC = gcc
 WRAPCC_CLANG = clang
 
-LDSO_PATHNAME = $(syslibdir)/ld-musl-$(ARCH)$(SUBARCH).so.1
+LDSO_PATHNAME = $(syslibdir)/ld-musl-$(MUSL_LDSO_ARCH).so.1
 
 export CONFIG = $(CURDIR)/config.mak
